test/dlc_tests/ec2/test_smdebug.py

```diff
@@ -10,73 +10,70 @@
-    if is_tf1(training):
-        pytest.skip("Currently skipping for TF1 until the issue is fixed")
-    run_smdebug_test(training, ec2_connection, region, docker_executable="nvidia-docker", container_name="smdebug-gpu",
-                     is_large_shm="p2.8xlarge" in SMDEBUG_EC2_GPU_INSTANCE_TYPE)
+def test_smdebug_gpu(training, ec2_connection, region, ec2_instance_type, gpu_only, py3_only):
+    run_smdebug_test(
+        training,
+        ec2_connection,
+        region,
+        ec2_instance_type,
+        docker_executable="nvidia-docker",
+        container_name="smdebug-gpu",
+    )
 
 
 @pytest.mark.flaky(reruns=0)
 @pytest.mark.integration("smdebug")
 @pytest.mark.model("mnist")
 @pytest.mark.parametrize("ec2_instance_type", SMDEBUG_EC2_CPU_INSTANCE_TYPE, indirect=True)
-def test_smdebug_cpu(training, ec2_connection, region, cpu_only, py3_only):
-    # TODO: Remove this once test timeout has been debugged (failures especially on m4.16xlarge)
-    if is_tf2(training) and "m4.16xlarge" in SMDEBUG_EC2_CPU_INSTANCE_TYPE:
-        pytest.skip("Currently skipping for TF2 on m4.16xlarge until the issue is fixed")
-    if is_tf1(training):
-        pytest.skip("Currently skipping for TF1 until the issue is fixed")
-    run_smdebug_test(training, ec2_connection, region)
+def test_smdebug_cpu(training, ec2_connection, region, ec2_instance_type, cpu_only, py3_only):
+    run_smdebug_test(training, ec2_connection, region, ec2_instance_type)
+
+
+class SMDebugTestFailure(Exception):
+    pass
 
 
 def run_smdebug_test(
     image_uri,
     ec2_connection,
     region,
+    ec2_instance_type,
     docker_executable="docker",
     container_name="smdebug",
     test_script=SMDEBUG_SCRIPT,
-    logfile="output.log",
-    is_large_shm=False
 ):
+    large_shm_instance_types = ("p2.8xlarge", "m4.16xlarge")
+    shm_setting = " --shm-size=1g " if ec2_instance_type in large_shm_instance_types else " "
     framework = get_framework_from_image_uri(image_uri)
     container_test_local_dir = os.path.join("$HOME", "container_tests")
     ec2_connection.run(f"$(aws ecr get-login --no-include-email --region {region})", hide=True)
 
-    shm_setting = '--shm-size=1g' if is_large_shm else ""
-    ec2_connection.run(
-        f"{docker_executable} run {shm_setting} --name {container_name} -v "
-        f"{container_test_local_dir}:{os.path.join(os.sep, 'test')} -itd {image_uri}",
-        hide=True,
-    )
-
     try:
-        test_output = ec2_connection.run(
-            f"{docker_executable} exec --user root {container_name} "
-            f"/bin/bash -c '{test_script} {framework}' | tee {logfile}",
+        ec2_connection.run(
+            f"{docker_executable} run --name {container_name} -v "
+            f"{container_test_local_dir}:{os.path.join(os.sep, 'test')}{shm_setting}{image_uri} "
+            f"./{test_script} {framework}",
             hide=True,
-            warn=True,
-            timeout=3000,
+            timeout=2400,
```

Comment:
30 mins seems like a more reasonable timeout - tested on small instance types (g3.4x and t2.2x) - tests still pass in this time frame.